src/cpu/flexcpu/inflight_inst.cc

```diff
@@ -0,0 +1,648 @@
+    // To make sure compiler doesn't complain
+    return backingContext->readVec8BitLaneReg(si->srcRegIdx(op_idx));
+    // TODO
+}
+
+ConstVecLane16
+InflightInst::readVec16BitLaneOperand(const StaticInst* si, int op_idx) const
+{
+    panic("readVec16BitLaneOperand() not implemented!");
+    // To make sure compiler doesn't complain
+    return backingContext->readVec16BitLaneReg(si->srcRegIdx(op_idx));
+    // TODO
+}
+
+ConstVecLane32
+InflightInst::readVec32BitLaneOperand(const StaticInst* si, int op_idx) const
+{
+    panic("readVec32BitLaneOperand() not implemented!");
+    // To make sure compiler doesn't complain
+    return backingContext->readVec32BitLaneReg(si->srcRegIdx(op_idx));
+    // TODO
+}
+
+ConstVecLane64
+InflightInst::readVec64BitLaneOperand(const StaticInst* si, int op_idx) const
+{
+    panic("readVec64BitLaneOperand() not implemented!");
+    // To make sure compiler doesn't complain
+    return backingContext->readVec64BitLaneReg(si->srcRegIdx(op_idx));
+    // TODO
+}
+
+
+void
+InflightInst::setVecLaneOperand(const StaticInst* si, int dst_idx,
+                        const LaneData<LaneSize::Byte>& val)
+{
+    panic("setVecLaneOperand() not implemented!");
+    // TODO
+}
+
+void
+InflightInst::setVecLaneOperand(const StaticInst* si, int dst_idx,
+                        const LaneData<LaneSize::TwoByte>& val)
+{
+    panic("setVecLaneOperand() not implemented!");
+    // TODO
+}
+
+void
+InflightInst::setVecLaneOperand(const StaticInst* si, int dst_idx,
+                        const LaneData<LaneSize::FourByte>& val)
+{
+    panic("setVecLaneOperand() not implemented!");
+    // TODO
+}
+
+void
+InflightInst::setVecLaneOperand(const StaticInst* si, int dst_idx,
+                        const LaneData<LaneSize::EightByte>& val)
+{
+    panic("setVecLaneOperand() not implemented!");
+    // TODO
+}
+
+VecElem
+InflightInst::readVecElemOperand(const StaticInst* si, int op_idx) const
+{
+    panic("readVecElemOperand() not implemented!");
+    return 0;
+    // TODO
+}
+
+void
+InflightInst::setVecElemOperand(const StaticInst* si, int dst_idx,
+                                const VecElem val)
+{
+    panic("setVecElemOperand() not implemented!");
+    // TODO
+}
+
+CCReg
+InflightInst::readCCRegOperand(const StaticInst* si, int op_idx)
+{
+    const RegId& reg_id = si->srcRegIdx(op_idx);
+    assert(reg_id.isCCReg());
+
+    const DataSource& source = sources[op_idx];
+    const shared_ptr<InflightInst> producer = source.producer.lock();
+
+    if (producer) {
+        // If the producing instruction is still in the buffer, grab the result
+        // assuming that it has been produced, and our index is in bounds.
+        assert(producer->isComplete());
+        assert(source.resultIdx >= 0
+            && source.resultIdx < producer->results.size()
+            && producer->resultValid[source.resultIdx]);
+        return producer->getResult(source.resultIdx).asCCReg();
```

Comment:
Definitely will keep a note on that there, but I don't know how simple it would be to capture specific bit reads/writes without doing the preexecution as we described before...

Reply:
And even then I'm not sure if it's simple to identify exactly which bits are being written by the instruction, if we want to rename based on bits.